asapdiscovery-modeling/asapdiscovery/modeling/tests/test_protein_prep.py

```diff
@@ -0,0 +1,284 @@
+
+        prot_fn = prepped_files / f"{xtal.output_name}-prepped_protein.pdb"
+        save_openeye_pdb(prot, str(prot_fn))
+
+        complex_fn = prepped_files / f"{xtal.output_name}-prepped_complex.pdb"
+        save_openeye_pdb(complex_, str(complex_fn))
+
+        lig_fn = prepped_files / f"{xtal.output_name}-prepped_ligand.sdf"
+        save_openeye_sdf(lig, str(lig_fn))
+
+        for fn in [prot_fn, complex_fn, lig_fn]:
+            assert Path(fn).exists()
+            assert Path(fn).is_file()
+
+        # TODO: Add a test to make sure the ligand is in the active site
+        # TODO: Add a test to make sure the seqres has been added to the protein
+
+    def test_mers_protein_prep(
+        self, mers_xtal, ref, prepped_files, loop_db, ref_chain="A"
+    ):
+        xtal = mers_xtal
+        # Load structure
+        prot = load_openeye_cif1(xtal.str_fn)
+
+        assert type(prot) == oechem.OEGraphMol
+
+        # Align to reference
+        prot = align_receptor(
+            prot,
+            dimer=True,
+            ref_prot=str(ref),
+            split_initial_complex=True,
+            split_ref=True,
+            ref_chain=ref_chain,
+            mobile_chain=xtal.active_site_chain,
+            keep_water=False,
+        )
+
+        save_openeye_pdb(prot, prepped_files / f"{xtal.output_name}_align.pdb")
+
+        # Mutate Residues
+        seqres_yaml = fetch_test_file("mpro_mers_seqres.yaml")
+
+        with open(seqres_yaml) as f:
+            seqres_dict = yaml.safe_load(f)
+        seqres = seqres_dict["SEQRES"]
+
+        res_list = seqres_to_res_list(seqres)
+
+        prot = mutate_residues(prot, res_list, place_h=True)
+        seqres = " ".join(res_list)
+
+        save_openeye_pdb(prot, prepped_files / f"{xtal.output_name}_mutate.pdb")
+
+        # Spruce Protein
+        du = spruce_protein(
+            initial_prot=prot,
+            seqres=seqres,
+            loop_db=loop_db,
+            return_du=True,
+            site_residue=xtal.active_site,
+        )
+        assert type(du) == oechem.OEDesignUnit
+
+        # Serialize output!
+
+        # TODO: Make sure this doesn't fail if there is no ligand
+        du_fn = prepped_files / f"{xtal.output_name}-prepped_receptor_0.oedu"
+        oechem.OEWriteDesignUnit(str(du_fn), du)
+
+        prot = oechem.OEGraphMol()
+        du.GetProtein(prot)
+        prot = add_seqres_to_openeye_protein(prot, seqres)
+
+        prot_fn = prepped_files / f"{xtal.output_name}-prepped_protein.pdb"
+        save_openeye_pdb(prot, str(prot_fn))
+
+        assert Path(prot_fn).exists()
+        assert Path(prot_fn).is_file()
+
+    @pytest.mark.parametrize(
+        ("du_fn", "has_lig"),
+        [
+            ("Mpro-P2660_0A_bound-prepped_receptor_0.oedu", True),
+            ("rcsb_8czv-assembly1-prepped_receptor_0.oedu", False),
+        ],
+    )
+    def test_openeye_du_loading(self, du_fn, has_lig, prepped_files):
```

Comment:
This tests these two specific files work, but would it be better to have a utility function to test the output of one of the prep_workflows themselves? Let me know if thats not clear.